pkg/cache/internal/informers.go

```diff
@@ -354,6 +360,9 @@ func (ip *Informers) addInformerToMap(gvk schema.GroupVersionKind, obj runtime.O
 		WatchFunc: func(opts metav1.ListOptions) (watch.Interface, error) {
 			ip.selector.ApplyToList(&opts)
 			opts.Watch = true // Watch needs to be set to true separately
+			if ip.watchTimeoutPeriod != nil {
+				opts.TimeoutSeconds = ptr.To(int64(ip.watchTimeoutPeriod.Seconds()))
```

Comment:
Since this is a certain time without any random, it might bring a large number of watch requests to apiserver when all the old watches closed and try to reconnect, if your operator watch a lot of resources as you described in this pr.

This is why the default timeout is a random number between 5 and 10 minutes. So how about add a little random time base on the given period?

Reply:
Thanks. That makes sense. Fixed following the style of https://github.com/kubernetes/client-go/blob/8c4efe8d079e405329f314fb789a41ac6af101dc/tools/cache/reflector.go#L418